pulsar-client-cpp/lib/PublisherStatsImpl.cc

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2016 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <lib/PublisherStatsImpl.h>
+#include <lib/LogUtils.h>
+
+namespace pulsar {
+DECLARE_LOG_OBJECT();
+
+    boost::array<double, 4> PublisherStatsImpl::probs = {0.5, 0.9,0.99,0.999};
```

Comment:
since ```probs``` is not part of the interface of this class, it should be a file scope static, not a class member. and const if the boost lib will allow that.